### PR TITLE
Separate inactive votes cache from gap cache

### PR DESCRIPTION
**Pull request by SergiySW**
~Also check 2 different conditions for confirmation and bootstrap start~
Functional changes were moved to different PR https://github.com/nanocurrency/nano-node/pull/2553. Now `inactive_votes_bootstrap_check ()` is just a copy of `nano::gap_cache::bootstrap_check ()`

### CODE REVIEW

File: nano/node/active_transactions.hpp [hunk 1]
```diff
@@ -181,8 +193,17 @@ class active_transactions final
...
 	static size_t constexpr max_priority_cementable_frontiers{ 100000 };
 	static size_t constexpr confirmed_frontiers_max_pending_cut_off{ 1000 };
-	nano::gap_cache::ordered_gaps inactive_votes_cache;
+	// clang-format off
+	using ordered_cache = boost::multi_index_container<nano::inactive_cache_information,
+	boost::multi_index::indexed_by<
+		boost::multi_index::ordered_non_unique<boost::multi_index::tag<tag_arrival>,
+			boost::multi_index::member<nano::inactive_cache_information, std::chrono::steady_clock::time_point, &nano::inactive_cache_information::arrival>>,
+		boost::multi_index::hashed_unique<boost::multi_index::tag<tag_hash>,
+			boost::multi_index::member<nano::inactive_cache_information, nano::block_hash, &nano::inactive_cache_information::hash>>>>;
```
Comment (guilhermelawless) [inline nano/node/active_transactions.hpp]: Could use the `mi` alias here.

File: nano/node/active_transactions.cpp [hunk 2]
```diff
@@ -959,45 +959,96 @@ void nano::active_transactions::add_inactive_votes_cache (nano::block_hash const
...
+	if (existing != inactive_votes_cache.get<tag_hash> ().end ())
+	{
+		inactive_votes_cache.get<tag_hash> ().erase (existing);
+	}
+}
+
+bool nano::active_transactions::inactive_votes_bootstrap_check (nano::inactive_cache_information const & info_a, bool & confirmed_a)
+{
+	uint128_t tally;
+	for (auto & voter : info_a.voters)
```
Comment (guilhermelawless) [inline nano/node/active_transactions.cpp]: Could be `auto const &`

File: nano/node/active_transactions.cpp [hunk 3]
```diff
@@ -943,11 +943,11 @@ void nano::active_transactions::add_inactive_votes_cache (nano::block_hash const
 	// Check principal representative status
 	if (node.ledger.weight (representative_a) > node.minimum_principal_weight ())
 	{
-		auto existing (inactive_votes_cache.get<nano::gap_cache::tag_hash> ().find (hash_a));
-		if (existing != inactive_votes_cache.get<nano::gap_cache::tag_hash> ().end () && !existing->confirmed)
+		auto existing (inactive_votes_cache.get<tag_hash> ().find (hash_a));
```
Comment (guilhermelawless) [inline nano/node/active_transactions.cpp]: I've been cleaning up these verbose `inactive_votes_cache.get<nano::gap_cache::tag_hash> ()` when they are used many times in a function, with:
```cpp
auto & inactive_by_hash (inactive_votes_cache.get<nano::gap_cache::tag_hash> ());
auto existing (inactive_by_hash.find (hash_a));
```

This is one of the places that is most verbose, so could consider applying here. Maybe whenever there are at least 2 uses, same for tag_arrival

File: nano/node/active_transactions.cpp [hunk 4]
```diff
@@ -1000,17 +1003,18 @@ nano::inactive_cache_information nano::active_transactions::find_inactive_votes_
...
 void nano::active_transactions::erase_inactive_votes_cache (nano::block_hash const & hash_a)
 {
-	auto existing (inactive_votes_cache.get<tag_hash> ().find (hash_a));
-	if (existing != inactive_votes_cache.get<tag_hash> ().end ())
+	auto & inactive_by_hash (inactive_votes_cache.get<tag_hash> ());
+	auto existing (inactive_by_hash.find (hash_a));
+	if (existing != inactive_by_hash.end ())
 	{
-		inactive_votes_cache.get<tag_hash> ().erase (existing);
+		inactive_by_hash.erase (hash_a);
```
Comment (guilhermelawless) [inline nano/node/active_transactions.cpp]: Maybe keep `existing` here to avoid re-hashing, otherwise LGTM

Comment (SergiySW) [reply nano/node/active_transactions.cpp]: Ah yes

File: nano/node/active_transactions.cpp [hunk 5]
```diff
@@ -1003,47 +1004,99 @@ void nano::active_transactions::add_inactive_votes_cache (nano::block_hash const
...
 						info.confirmed = true;
 					});
 				}
 			}
 		}
 		else
 		{
-			inactive_votes_cache.get<nano::gap_cache::tag_arrival> ().emplace (nano::gap_information{ std::chrono::steady_clock::now (), hash_a, std::vector<nano::account> (1, representative_a) });
+			std::vector<nano::account> representative_vector (1, representative_a);
+			auto start_bootstrap (inactive_votes_bootstrap_check (representative_vector, hash_a));
```
Comment (guilhermelawless) [inline nano/node/active_transactions.cpp]: This is a functional change right?

File: nano/node/active_transactions.cpp [hunk 6]
```diff
@@ -1003,47 +1004,99 @@ void nano::active_transactions::add_inactive_votes_cache (nano::block_hash const
 
 			if (is_new)
 			{
-				if (node.gap_cache.bootstrap_check (existing->voters, hash_a))
+				bool confirmed (false);
```
Comment (guilhermelawless) [inline nano/node/active_transactions.cpp]: Seems leftover

Comment (SergiySW) [reply nano/node/active_transactions.cpp]: Removed

File: nano/node/active_transactions.cpp (hunk 5 above)
Comment (SergiySW) [reply nano/node/active_transactions.cpp]: Moved to new PR only

**Pull request merged**